### ISSUE TITLE
[SLP] "Deleting out-of-tree value" assertion triggers in SLP vectorizer

### ISSUE THREAD

**Issue by momo5502** [llvm/llvm-project#95925]
The following sample triggers the "Deleting out-of-tree value" assertion in SLP vectorizer:

```LLVM
target datalayout = "e-m:w-p270:32:32-p271:32:32-p272:64:64-i64:64-i128:128-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-windows-msvc19.39.33523"

define fastcc ptr @"testfunc"(ptr %0) {
entry:
  %1 = ptrtoint ptr %0 to i64
  %2 = trunc i64 %1 to i32
  %3 = ptrtoint ptr null to i64
  %4 = trunc i64 %3 to i32
  switch i32 0, label %newFuncRoot994 [
    i32 1, label %newFuncRoot994
    i32 0, label %newFuncRoot584
  ]

newFuncRoot584:                                   ; preds = %entry
  %5 = insertelement <4 x i32> poison, i32 %4, i64 0
  %6 = insertelement <4 x i32> %5, i32 %2, i64 1
  ret ptr null

newFuncRoot994:                                   ; preds = %entry, %entry
  %7 = insertelement <4 x i32> zeroinitializer, i32 %2, i64 %3
  ret ptr null
}
```

https://godbolt.org/z/4j9fonn1d

**Comment by momo5502** [comment]
I am investigating this issue myself. Not sure if it's possible to assign it to me :D

**Comment by momo5502** [comment]
As I really need the fix, I'm gonna create a PR that removes the continue. If that is not a viable solution, the PR could be closed or adjusted according to the review comments.

**Comment by alexey-bataev** [comment]
Fixed in 0280f97b36c83a7129e5dbce53c603b7ec5d82fe
